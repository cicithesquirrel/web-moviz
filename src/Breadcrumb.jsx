import React from 'react';

const Breadcrumb = ({crumbs}) => (
    <ol className="breadcrumb">
        {crumbs.map((c)=> (<li key={c.label}><a href={(c.link?c.link:'#')}>{c.label}</a></li>))}
    </ol>
);

export default Breadcrumb;