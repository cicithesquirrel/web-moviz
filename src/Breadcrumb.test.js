import React from 'react';
import Breadcrumb from './Breadcrumb';
import renderer from 'react-test-renderer';

test('Breadcrumb is empty', () => {
  const component = renderer.create(
    <Breadcrumb crumbs={[]} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Breadcrumb with label only', () => {
  const component = renderer.create(
    <Breadcrumb crumbs={[{label:'blabla'}]} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Breadcrumb with label and link', () => {
  const component = renderer.create(
    <Breadcrumb crumbs={[{label:'blabla', link:'link_to_blabla'}]} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Breadcrumb with several items', () => {
  const component = renderer.create(
    <Breadcrumb crumbs={[{label:'blabla', link:'link_to_blabla'}, {label:'blabla1', link:'link_to_blabla1'}, {label:'blabla2', link:'link_to_blabla2'}]} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
