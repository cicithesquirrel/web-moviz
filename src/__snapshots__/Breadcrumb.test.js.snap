exports[`test Breadcrumb is empty 1`] = `
<ol
  className="breadcrumb" />
`;

exports[`test Breadcrumb with label and link 1`] = `
<ol
  className="breadcrumb">
  <li>
    <a
      href="link_to_blabla">
      blabla
    </a>
  </li>
</ol>
`;

exports[`test Breadcrumb with label only 1`] = `
<ol
  className="breadcrumb">
  <li>
    <a
      href="#">
      blabla
    </a>
  </li>
</ol>
`;

exports[`test Breadcrumb with several items 1`] = `
<ol
  className="breadcrumb">
  <li>
    <a
      href="link_to_blabla">
      blabla
    </a>
  </li>
  <li>
    <a
      href="link_to_blabla1">
      blabla1
    </a>
  </li>
  <li>
    <a
      href="link_to_blabla2">
      blabla2
    </a>
  </li>
</ol>
`;
